code/modules/games/cas.dm

```diff
@@ -0,0 +1,120 @@
+
+/obj/item/toy/cards/deck/cas/blanks
+	name = "\improper CAS deck (white blanks)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is a white deck which contains blanks to be used in custom games of CAS."
+	decksize = 0
+	blanks = 50
+
+/obj/item/toy/cards/deck/cas/black
+	name = "\improper CAS deck (black)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is the black deck."
+	icon_state = "deck_casblack_full"
+	deckstyle = "casblack"
+	blanks = 0
+	decksize = 50
+	card_text_file = "strings/cas_black.txt"
+
+/obj/item/toy/cards/deck/cas/black/blanks
+	name = "\improper CAS deck (black blanks)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is a black deck which contains blanks to be used in custom games of CAS."
+	decksize = 0
+	blanks = 50
+
+/obj/item/toy/cards/deck/cas/New()
+	allcards = file2list(card_text_file)
+	var/list/possiblecards = allcards.Copy()
+	if(possiblecards.len < decksize) // sanity check
+		decksize = (possiblecards.len - 1)
+	var/list/randomcards = list()
+	while (randomcards.len < decksize)
+		randomcards += pick_n_take(possiblecards)
+	for(var/i=1 to randomcards.len)
+		var/cardtext = randomcards[i]
+		cards += "[cardtext]"
+	if(!blanks)
+		return
+	for(var/x=1 to blanks)
+		cards += "Blank Card"
+
+/obj/item/toy/cards/deck/cas/attack_hand(mob/user)
+	if(user.lying)
+		return
+	var/choice = null
+	if(cards.len == 0)
+		user << "<span class='warning'>There are no more cards to draw!</span>"
+		return
+	var/obj/item/toy/cards/singlecard/cas/H = new/obj/item/toy/cards/singlecard/cas(user.loc)
+	choice = cards[1]
+	H.name = "[copytext(deckstyle,4)] [H.name]"
+	if(deckstyle == "caswhite")
+		H.pixel_x = 5
+	H.cardname = choice
+	if(choice == "Blank Card")
+		H.blank = 1
+	H.parentdeck = src
+	H.icon_state = "singlecard_[deckstyle]"
+	src.cards -= choice
+	H.pickup(user)
+	user.put_in_hands(H)
+	user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
+	update_icon()
+
+/obj/item/toy/cards/deck/cas/update_icon()
+	if(cards.len < 26)
+		icon_state = "deck_[deckstyle]_low"
+
+/obj/item/toy/cards/singlecard/cas
+	name = "CAS card"
+	desc = "A CAS card."
+	icon_state = "cas_white"
+	var/blank = 0
+
+/obj/item/toy/cards/singlecard/cas/examine(mob/user)
+	if(ishuman(user))
+		if(in_range(user, src))
+			user << "<span class='notice'>The card reads: [src.cardname]</span>"
+		else
+			user << "<span class='warning'>You need to get closer to check the card!</span>"
+
+/obj/item/toy/cards/singlecard/cas/Flip()
+	return
+
+/obj/item/toy/cards/singlecard/cas/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/weapon/pen))
+		if(!blank)
```

Comment:
this blank var is a bit pointless,you could just use istype(I, whatevertheblankcardpathis)